src/main/java/com/rapid7/client/dcerpc/mslsad/objects/SAMPRDomainPasswordInfo.java

```diff
@@ -0,0 +1,160 @@
+     * | 0x00000002                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_NO_CLEAR_CHANGE | Change-password methods that provide the cleartext password are disabled |
+     * | 0x00000004                      | by the server.                                                           |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_LOCKOUT_ADMINS           | Reserved. No effect on password policy.                                  |
+     * | 0x00000008                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_STORE_CLEARTEXT | The server MUST store the cleartext password, not just the computed      |
+     * | 0x00000010                      | hashes.                                                                  |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_REFUSE_PASSWORD_CHANGE   | Reserved. No effect on password policy.                                  |
+     * | 0x00000020                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     */
+    private short minimumPasswordLength;
+    private short passwordHistoryLength;
+    private int passwordProperties;
+    private int maximumPasswordAge;
+    private int minimumPasswordAge;
+
+    public boolean isDomainPasswordComplex() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_COMPLEX) != 0;
+    }
+
+    public boolean isDomainPasswordNoAnonChange() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_NO_ANON_CHANGE) != 0;
+    }
+
+    public boolean isDomainPasswordStoredClearText() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_STORE_CLEARTEXT) != 0;
+    }
+
+    public short getMinimumPasswordLength() {
+        return minimumPasswordLength;
+    }
+
+    public short getPasswordHistoryLength() {
+        return passwordHistoryLength;
+    }
+
+    public int getPasswordProperties() {
+        return passwordProperties;
+    }
+
+    public int getMaximumPasswordAge() {
+        return maximumPasswordAge;
+    }
+
+    public int getMinimumPasswordAge() {
+        return minimumPasswordAge;
+    }
+
+    @Override
+    public void unmarshalPreamble(PacketInput in) throws IOException {
+    }
+
+    @Override
+    public void unmarshalEntity(PacketInput in) throws IOException {
+        this.minimumPasswordLength = in.readShort();
```

Comment:
You should align here as per the new alignment strategy:
```
@Override
public void unmarshalEntity(PacketInput in) throws IOException {
    // Structure Alignment: 4
    in.align(Alignment.FOUR);
```

This structure's alignment is 4:
```
USHORT        MinPasswordLength;: 2
USHORT        PasswordHistoryLength;: 2
ULONG         PasswordProperties;: 4
LARGE_INTEGER MaxPasswordAge;: 4 (it's an 8 byte long, but comprised of 2x4 byte integers)
LARGE_INTEGER MinPasswordAge;: 4 (it's an 8 byte long, but comprised of 2x4 byte integers)
```